File: backend/routers/trading.py
from fastapi import APIRouter, HTTPException
from typing import List
from models import MarketItem

router = APIRouter()

# Mock data
mock_market_items = [
    MarketItem(
        id=1,
        name="Legendary Nexus Warrior",
        price=150.00,
        currency="DAR",
        seller="Trader123",
        rarity="Legendary"
    ),
    MarketItem(
        id=2,
        name="Epic Stellar Mage",
        price=75.50,
        currency="DAR",
        seller="ProSeller",
        rarity="Epic"
    )
]

@router.get("/markets", response_model=List[MarketItem])
async def get_market_items():
    """Get all market items"""
    return mock_market_items

@router.get("/markets/{item_id}", response_model=MarketItem)
async def get_market_item(item_id: int):
    """Get a specific market item by ID"""
    for item in mock_market_items:
        if item.id == item_id:
            return item
    raise HTTPException(status_code=404, detail="Item not found")

@router.get("/buy_token")
async def get_token_info():
    """Get information about buying $D token"""
    return {
        "token_name": "Dalarnia Token",
        "symbol": "DAR",
        "exchanges": [
            {"name": "Binance", "url": "#"},
            {"name": "KuCoin", "url": "#"},
            {"name": "Gate.io", "url": "#"}
        ],
        "current_price": "$0.15",
        "market_cap": "$45M"
    }

File: frontend/src/data/trading.ts
import { MarketItem, TokenInfo } from "@/types";

export const marketItems: MarketItem[] = [
  {
    id: "item-1",
    name: "Riftblade Blueprint",
    price: 420,
    currency: "DAR",
    rarity: "Legendary",
    seller: "LyraLabs",
  },
  {
    id: "item-2",
    name: "Magma Core Shield",
    price: 180,
    currency: "DAR",
    rarity: "Epic",
    seller: "ForgeFrontier",
  },
  {
    id: "item-3",
    name: "Scout Drone Mk.II",
    price: 95,
    currency: "DAR",
    rarity: "Rare",
    seller: "Pulseworks",
  },
  {
    id: "item-4",
    name: "Verdant Bloom Seeds",
    price: 45,
    currency: "DAR",
    rarity: "Common",
    seller: "Greenline",
  },
];

export const tokenInfo: TokenInfo = {
  tokenName: "Dalarnia Token",
  symbol: "DAR",
  currentPrice: "$0.18",
  marketCap: "$52M",
  exchanges: [
    { name: "Kraken", url: "https://kraken.com" },
    { name: "Binance", url: "https://binance.com" },
    { name: "OKX", url: "https://okx.com" },
  ],
};
